src/utils/storageFacade.utils.ts

```diff
@@ -1,8 +1,94 @@
 import { supabase } from './supabase';
-import { Package, Vulnerability } from './types.utils';
+import {
+  Package,
+  Vulnerability,
+  PackageViewParam,
+  VulnerabilityViewParam,
+  DisplayPackage,
+  severityRating,
+} from './types.utils';
 import { logger } from './logger.utils';
 
-/****************** PUBLIC API **********************/
+const PAGE_SIZE = 25;
+
+/****************** PUBLIC API: SYSTEM **********************/
+export async function countPackages(sessionid: number) {
+  let { count, error } = await supabase //common syntax on JS: const {data,error} = await...
+    .from('packages')
+    .select('*', { count: 'exact' }) //values are outputted first in, last out
+    .eq('sessionid', sessionid);
+
+  if (error) {
+    logger.error(error);
+  }
+  if (count) {
+    return count;
+  } else {
+    return 0;
+  }
+}
+
+export async function countVulnerabilities(sessionid: number) {
+  let { count, error } = await supabase
+    .from('vulnerabilities')
+    .select(
+      '*,junction!inner(packageid,packages!inner(sessionid, package_ref))',
+      { count: 'exact' }
+    )
+    .eq('junction.packages.sessionid', sessionid);
+  if (error) {
+    logger.error(error);
+  }
+  if (count) {
+    return count;
+  } else {
+    return 0;
+  }
+}
+
+export async function countHighSeverityCves(sessionid: number) {
+  let { count, error } = await supabase
+    .from('vulnerabilities')
+    .select(
+      '*,junction!inner(packageid,packages!inner(sessionid, package_ref))',
+      { count: 'exact' }
+    )
+    .eq('junction.packages.sessionid', sessionid)
+    .gte('severity', severityRating.HIGH);
```

Comment:
put a note here that this counts high and critical severity - may not be obvious from the name of the function that this includes critical also